### ISSUE TITLE
SC2329 is incorrectly shown if sourced file contains "return"

### ISSUE THREAD

**Issue by jansorg** [koalaman/shellcheck#3298]
#### For bugs
- Rule Id: SC2329
- My shellcheck version (`shellcheck --version` or 'online'): 0.11.0

#### Here's a snippet or screenshot that shows the problem:

This setup has two files: `main.bash` which sources `library.bash`.

- File `library.bash`:
```sh
#!/bin/bash

return 0
```

- File `main.bash`:
```sh
#!/bin/bash

# shellcheck source=SCRIPTDIR/library.bash
. "$PWD/library.bash"

function say_hello() {
  echo hello
}

say_hello
```

#### Here's what shellcheck currently says:
```
$ shellcheck -x /home/jansorg/temp/bash-test/library/main.bash

In /home/jansorg/temp/bash-test/library/main.bash line 6:
function say_hello() {
^-- SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).
^-- SC2329 (info): This function is never invoked. Check usage (or ignored if invoked indirectly).


In /home/jansorg/temp/bash-test/library/main.bash line 10:
say_hello
^-------^ SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

For more information:
  https://www.shellcheck.net/wiki/SC2317 -- Command appears to be unreachable...
  https://www.shellcheck.net/wiki/SC2329 -- This function is never invoked. C...
```

#### Here's what I wanted or expected to see:

It shouldn't warn about SC2317 and not about SC2329.

Executing the file with bash shows output `hello` as expected.

`man bash` says that a `return` in a sourced file sets the status code of the `source` command, it does not terminate the main script:

>   source [-p path] filename [arguments]
     The . command (source) reads and execute commands from filename in the current shell environment  and
     returns the exit status of the last command executed from filename.
